ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/review/v1/review/clientlibs/site/js/reviewview.js

```diff
@@ -0,0 +1,320 @@
+          const children = this.#renderReviewFields(panels);
+          reviewContainer.innerHTML = '';
+          reviewContainer.appendChild(children);
+        }
+      })
+    }
+
+    // iterate over all the panels or linked panels and render child of each panel
+    #renderReviewFields(items) {
+      if (!items) return;
+      const currentFragment = document.createDocumentFragment();
+      items.filter(item => item.visible !== false && item.fieldType && item[':type']).forEach(item => {
+        if (Review.isRepeatable(item)) {
+          const repeatablePanel = this.#renderReviewFields(item.items);
+          if (Review.hasChild(repeatablePanel)) {
+            currentFragment.appendChild(repeatablePanel);
+          }
+        } else {
+          let template = this.templateMappings[item.fieldType] || this.templateMappings['default'];
+          const cloneNode = template.content.cloneNode(true);
+          Review.addClassModifier(cloneNode, item);
+          Review.renderLabel(cloneNode, item);
+          this.#renderEditButton(cloneNode, item);
+          if (item.fieldType === 'panel') {
+            const fields = this.#renderReviewFields(item.items);
+            if (Review.hasChild(fields)) {
+              const labelContainer = cloneNode.querySelector(`.${Review.bemBlock}__label-container`);
+              if(item?.label?.visible === false && labelContainer){
+                labelContainer.remove();
+              }
+              Review.addAccessibilityAttributes(cloneNode, item);
+              const contentElement = cloneNode.querySelector('[' + Review.templateAttribute + '-content]');
+              if(contentElement){
+                contentElement.appendChild(fields);
+              }
+              currentFragment.appendChild(cloneNode);
+            }
+          } else if (item.fieldType !== 'button' && item.fieldType !== 'plain-text' && !item[':type'].endsWith('review')) {
+            Review.renderValue(cloneNode, item);
+            Review.addAccessibilityAttributes(cloneNode, item);
+            currentFragment.appendChild(cloneNode);
+          }
+        }
+      });
+      return currentFragment;
+    }
+
+    #getPanels() {
+      const linkedPanels = this._model?._jsonModel?.properties["fd:linkedPanels"] || [];
+      if (linkedPanels.length) {
+        return this.#getLinkedPanels(linkedPanels);
+      }
+      return this.#getAllPanels();
+    }
+
+    // return all top lavel panels if author has not linked any panel
+    #getAllPanels() {
+      let state = this.formContainer._model.getState();
+      while (state?.items?.length === 1) {
+        state = state.items[0];
+      }
+      return state.items || [];
+    }
+
+    // return linked panels if author has linked any panel
+    #getLinkedPanels(linkedPanels) {
+      let queue = [], result = [];
+      let state = this.formContainer._model.getState();
+      queue.push(state);
+      while (queue.length && linkedPanels.length) {
```

Comment:
Why should caller understand what qualified name is ? Ideally, we need to use the directly exposed af-core API to get the model, rather than walking through the internals of formContainer

please note this.formContainer._model is internal, we should not access a private variable here

Reply:
changed